05-JWT-Basics/starter/middleware/auth.js

```diff
@@ -0,0 +1,20 @@
+const jwt = require("jsonwebtoken")
+const UnauthenticatedError = require("../errors/unauthenticated")
+
+const authenticationMiddleware = async (req, res, next) => {
+  const authHeader = req.headers.authorization
+  if (!authHeader || !authHeader.startsWith("Bearer")) {
+    throw new UnauthenticatedError("No token provided")
+  }
+  const token = authHeader.split(" ")[1]
+  try {
+    const decoded = jwt.verify(token, process.env.JWT_SECRET)
```

Comment:
Check for the presence or absence of `JWT_SECRET` during startup. Fail immediately if it's missing.

Reply:
I'm throwing a new error when starting the app if there is no JWT_SECRET.